web/models/announcement.ex

```diff
@@ -44,6 +44,12 @@ defmodule Constable.Announcement do
     ]
   end
 
+  def without_interests(query, interests \\ []) do
+    from a in query,
+      full_join: i in assoc(a, :interests),
+      where: not i.name in ^interests or is_nil(i.name)
```

Comment:
I'm a bit confused by the `is_nil(i.name)`. Is that saying it should also exclude announcements that on't have an interest at all?